cmd/csi-snapshotter/main.go

```diff
@@ -62,6 +62,7 @@ var (
 	showVersion            = flag.Bool("version", false, "Show version.")
 	threads                = flag.Int("worker-threads", 10, "Number of worker threads.")
 	csiTimeout             = flag.Duration("timeout", defaultCSITimeout, "The timeout for any RPCs to the CSI driver. Default is 1 minute.")
+	extraCreateMetadata    = flag.Bool("extra-create-metadata", false, "If set, add snapshot metadata to plugin snapshot requests as parameters.")
```

Comment:
IMO, this is a per-snapshot-class setting rather than controller level toggle, and here is why:
In a situation where there are more than one (say two) csi drivers in a k8s cluster.
CSI-A needs these three parameters, CSI-B does not need (or even worse it would fail if non-recognizable parameters have been passed in), then having a controller level toggle will cause CSI-B not working.

My take is to have predefined keys defined in controller, but require their existences in "parameters" of VolumeSnapshot class for the controller to add those extra parameters at runtime.

Reply:
@yuxiangqian the csi-snapshotter is run per csi driver. I don't see why this needs to be per SnapshotClass.

Reply:
please ignore this comment as its irrelevant. I thought this flag was in the snapshot controller.